crates/biome_js_analyze/src/analyzers/nursery/no_done_callback.rs

```diff
@@ -0,0 +1,168 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsArrowFunctionParameters, AnyJsBinding, AnyJsExpression, JsCallExpression, JsParameters,
+    JsTemplateExpression,
+};
+use biome_rowan::{AstNode, TextRange};
+
+declare_rule! {
+    /// This rule checks the function parameter of hooks & tests for use of the done argument, suggesting you return a promise instead.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// beforeEach(done => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// test('myFunction()', done => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// beforeEach(async () => {
+    ///     // ...
+    /// });
+    /// ```
+    ///
+    /// ```js
+    /// test('myFunction()', () => {
+    ///     expect(myFunction()).toBeTruthy();
+    /// });
+    /// ```
+    ///
+    pub NoDoneCallback {
+        version: "next",
+        name: "noDoneCallback",
+        recommended: true,
+        source: RuleSource::EslintJest("no-done-callback"),
+        source_kind: RuleSourceKind::Inspired,
+    }
+}
+
+impl Rule for NoDoneCallback {
+    type Query = Ast<JsCallExpression>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let callee = node.callee().ok()?;
+
+        let mut is_test_each = false;
+        let callee_member = callee.get_callee_member_name();
+        if let Some(member_name) = callee_member {
+            if member_name.text_trimmed() == "each" {
+                is_test_each = true;
+            }
+        }
```

Comment:
We can refactor like this
```suggestion
        let is_test_each = callee
            .get_callee_member_name()
            .map_or(false, |m| m.text_trimmed() == "each");
```